sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -208,18 +226,28 @@ def send_batch(self, event_data_batch, **kwargs):
                 :caption: Sends event data
 
         """
+        partition_id = kwargs.get("partition_id")
+        partition_key = kwargs.get("partition_key")
+        if isinstance(event_data_batch, EventDataBatch):
+            if partition_id or partition_key:
+                raise TypeError("partition_id and partition_key should be None when sending an EventDataBatch "
```

Comment:
Curious why we raise but not ignoring them?

Reply:
pk and pid are important information when sending an event data. It's better let them know if they provide wrong data.